src/stream/src/from_proto/group_top_n.rs

```diff
@@ -42,37 +45,100 @@ impl ExecutorBuilder for GroupTopNExecutorBuilder {
-                1 << 16,
-            )?
-            .boxed())
+                lru_manager: stream.context.lru_manager.clone(),
+                cache_size: 1 << 16,
+                with_ties: true,
+                group_key_types,
+            };
+            args.dispatch()
         } else {
-            Ok(GroupTopNExecutor::new_without_ties(
-                params.input.remove(0),
-                params.actor_context,
+            let args = GroupTopNExecutorDispatcherArgs {
+                input,
+                ctx: params.actor_context,
                 order_pairs,
-                (node.offset as usize, node.limit as usize),
-                node.order_by_len as usize,
-                params.pk_indices,
-                params.executor_id,
+                offset_and_limit: (node.offset as usize, node.limit as usize),
+                order_by_len: node.order_by_len as usize,
+                pk_indices: params.pk_indices,
+                executor_id: params.executor_id,
                 group_by,
                 state_table,
-                stream.context.lru_manager.clone(),
-                1 << 16,
+                lru_manager: stream.context.lru_manager.clone(),
+                cache_size: 1 << 16,
+                with_ties: false,
+                group_key_types,
+            };
+            args.dispatch()
+        }
+    }
+}
+
+struct GroupTopNExecutorDispatcherArgs<S: StateStore> {
+    input: BoxedExecutor,
+    ctx: ActorContextRef,
+    order_pairs: Vec<OrderPair>,
+    offset_and_limit: (usize, usize),
+    order_by_len: usize,
+    pk_indices: PkIndices,
+    executor_id: u64,
+    group_by: Vec<usize>,
+    state_table: StateTable<S>,
+    lru_manager: Option<LruManagerRef>,
+    cache_size: usize,
+    with_ties: bool,
+    group_key_types: Vec<DataType>,
+}
+
+impl<S: StateStore> HashKeyDispatcher for GroupTopNExecutorDispatcherArgs<S> {
+    type Output = StreamResult<BoxedExecutor>;
+
+    fn dispatch_impl<K: HashKey>(self) -> Self::Output {
+        match self.with_ties {
+            true => Ok(GroupTopNExecutor::<K, S, true>::new_with_ties(
+                self.input,
+                self.ctx,
+                self.order_pairs,
+                self.offset_and_limit,
+                self.order_by_len,
+                self.pk_indices,
+                self.executor_id,
+                self.group_by,
+                self.state_table,
+                self.lru_manager,
+                self.cache_size,
+            )?
+            .boxed()),
+            false => Ok(GroupTopNExecutor::<K, S, false>::new_without_ties(
+                self.input,
+                self.ctx,
+                self.order_pairs,
+                self.offset_and_limit,
+                self.order_by_len,
+                self.pk_indices,
+                self.executor_id,
+                self.group_by,
+                self.state_table,
+                self.lru_manager,
+                self.cache_size,
```

Comment:
IMO not very necessary

Reply:
BTW, the reason why I created two functions `new_with_ties` and `new_without_ties` is to help type inference, i.e., no need to `::<>` (and thus easier to refactor at that time). For now it seems we can merge them into one since we will alwasy have `::<K, ..>` 🤣

Reply:
Let me merge `new_with_ties` and `new_without_ties`  in this PR.